internal/handlers/handlers.go

```diff
@@ -0,0 +1,67 @@
+package handlers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/gorilla/mux"
+	"golang.org/x/oauth2"
+)
+
+var (
+	googleOauthConfig = &oauth2.Config{
+		ClientID:     os.Getenv("GOOGLE_CLIENT_ID"),
+		ClientSecret: os.Getenv("GOOGLE_CLIENT_SECRET"),
+		RedirectURL:  "http://localhost:3000/auth/google/callback",
+		Scopes:       []string{"https://www.googleapis.com/auth/plus.login"},
+		Endpoint: oauth2.Endpoint{
+			AuthURL:  "https://accounts.google.com/o/oauth2/auth",
+			TokenURL: "https://accounts.google.com/o/oauth2/token",
+		},
+	}
+)
+
+func handleGoogleAuth(w http.ResponseWriter, r *http.Request) {
+	url := googleOauthConfig.AuthCodeURL("state", oauth2.AccessTypeOffline)
+	http.Redirect(w, r, url, http.StatusTemporaryRedirect)
+}
+
+func handleGoogleCallback(w http.ResponseWriter, r *http.Request) {
+
+	code := r.FormValue("code")
```

Comment:
Should we have an `if` branch to make sure this exists? If not, respond with an error type thing.